R/tostring.R

```diff
@@ -103,24 +105,78 @@ setMethod("toString", "MatrixPrintForm", function(x,
                                                   tf_wrap = FALSE,
                                                   max_width = NULL,
                                                   col_gap = x$col_gap,
-                                                  hsep = default_hsep()) {
-  mat <- matrix_form(x, TRUE)
+                                                  hsep = default_hsep(),
+                                                  indent_size = NULL) {
+
+  checkmate::assert_integerish(indent_size, null.ok = TRUE)
+  checkmate::assert_flag(tf_wrap)
+
+  mat <- matrix_form(x, indent_rownames = TRUE) # Fix for indent_size should be here
   inset <- table_inset(mat)
 
+  # Standard indentation size and inserted one
+  # NB: Header can be avoided here because it is not affected by size (?)
+  if (!is.null(indent_size) && (mat$indent_size != indent_size)) {
+    diff_indent_size <- indent_size - mat$indent_size
+    # Add space if inserted indentation_size is larger
+    if (diff_indent_size > 0) {
+      mat$strings[-seq_len(mf_nlheader(mat)), 1] <-
+        paste(
+          sapply(mf_rinfo(mat)$indent, function(i) {
+            paste0(rep(" ", i * diff_indent_size), collapse = "")
+          }),
+          mat$strings[-seq_len(mf_nlheader(mat)), 1]
+        )
+      # formats
+      mat$formats[-seq_len(mf_nlheader(mat)), 1] <-
+        paste(
+          sapply(mf_rinfo(mat)$indent, function(i) {
+            paste0(rep(" ", i * diff_indent_size), collapse = "")
+          }),
+          mat$formats[-seq_len(mf_nlheader(mat)), 1]
+        )
+    } else {
+      # Take out a space if inserted indentation is smaller
```

Comment:
not working as expected. Still to correct for negative cases

Reply:
Negative indents have never been supported, minimum indent is 0. We should probably catch such things as an error in various places but that should be filed as a separate issue.

Reply:
eheh yes, this was a sort of relative indentation. It does not make a ton of sense now. I will add the check for indent_size > 0